src/header-generator.js

```diff
@@ -0,0 +1,271 @@
+const fs = require('fs');
+const path = require('path');
+const { BayesianNetwork } = require('bayesian-network');
+const { default: ow } = require('ow');
+
+const headerNetworkDefinitionPath = path.join(__dirname, './headerNetworkDefinition.json');
+const inputNetworkDefinitionPath = path.join(__dirname, './inputNetworkDefinition.json');
+const browserHelperFilePath = path.join(__dirname, './browserHelperFile.json');
+
+const browserHttpNodeName = '*BROWSER_HTTP';
+const operatingSystemNodeName = '*OPERATING_SYSTEM';
+const deviceNodeName = '*DEVICE';
+const missingValueDatasetToken = '*MISSING_VALUE*';
+
+const http2SecFetchAttributes = {
+    mode: 'sec-fetch-mode',
+    dest: 'sec-fetch-dest',
+    site: 'sec-fetch-site',
+    user: 'sec-fetch-user',
+};
+
+const http1SecFetchAttributes = {
+    mode: 'Sec-Fetch-Mode',
+    dest: 'Sec-Fetch-Dest',
+    site: 'Sec-Fetch-Site',
+    user: 'Sec-Fetch-User',
+};
+
+function getRandomInteger(minimum, maximum) {
+    return minimum + Math.floor(Math.random() * (maximum - minimum + 1));
+}
+
+function shuffleArray(array) {
+    if (array.length > 1) {
+        for (let x = 0; x < 10; x++) {
+            const position1 = getRandomInteger(0, array.length - 1);
+            const position2 = getRandomInteger(0, array.length - 1);
+            const holder = array[position1];
+            array[position1] = array[position2];
+            array[position2] = holder;
+        }
+    }
+
+    return array;
+}
+
+function browserVersionIsLesserOrEquals(browserVersionL, browserVersionR) {
+    return browserVersionL[0] <= browserVersionR[0];
+}
+
+function prepareHttpBrowserObject(httpBrowserString) {
+    const [browserString, httpVersion] = httpBrowserString.split('|');
+    const browserObject = browserString === missingValueDatasetToken ? { name: missingValueDatasetToken } : prepareBrowserObject(browserString);
+    return {
+        ...browserObject,
+        ...{
+            httpVersion,
+            completeString: httpBrowserString,
+        },
+    };
+}
+
+function prepareBrowserObject(browserString) {
+    const nameVersionSplit = browserString.split('/');
+    const versionSplit = nameVersionSplit[1].split('.');
+    const preparedVersion = [];
+    for (const versionPart of versionSplit) {
+        preparedVersion.push(parseInt(versionPart, 10));
+    }
+
+    return {
+        name: nameVersionSplit[0],
+        version: preparedVersion,
+        completeString: browserString,
+    };
+}
+
+const headerGeneratorOptionsShape = {
+    browsers: ow.optional.array,
+    operatingSystems: ow.optional.array,
```

Comment:
I think we should validate also the contents of the array.

Reply:
Wasn't sure how to do that without looping over the items and validating them one by one. Will probably do the looping option, let me know if you have any better idea.

Reply:
I think we can use this https://sindresorhus.com/ow/classes/arraypredicate.html#validate, or custom validator. There is an example from SDK, but there are no arrays. https://github.com/apify/apify-js/blob/9f94909103b801419ee6b367f85e5b698b7c2530/src/validators.js#L9